File: CS307-Database Design/Project1/station.py
import psycopg2
import json
from datetime import datetime

# 连接到 PostgreSQL 数据库
conn = psycopg2.connect(
    dbname="project1",
    user="postgres",
    password="123456789Bw&",
    host="localhost",
    port="5432"
)
cursor = conn.cursor()

cursor.execute("DELETE FROM station")

# 打开 JSON 文件并插入数据
with open('src/stations.json', 'r', encoding='utf-8') as f:
    data = json.load(f)

    # 插入数据
    for item in data.keys():
        station_name = item
        district = data[item]['district']
        intro = data[item]['intro']
        chinese_name = data[item]['chinese_name']

        # 执行插入操作
        cursor.execute(
            "INSERT INTO station (station_name,district,intro,chinese_name) VALUES (%s, %s, %s,%s)",
            (station_name,district,intro,chinese_name))


# 提交事务
conn.commit()

# 关闭连接
cursor.close()
conn.close()

File: CS307-Database Design/Project1/Station.java
public class Station {
    private String stationName;
    private String district;
    private String intro;
    private String chineseName;

    // 构造方法
    public Station(String stationName, String district, String intro, String chineseName) {
        this.stationName = stationName;
        this.district = district;
        this.intro = intro;
        this.chineseName = chineseName;
    }

    // Getter 和 Setter 方法
    public String getStationName() {
        return stationName;
    }

    public void setStationName(String stationName) {
        this.stationName = stationName;
    }

    public String getDistrict() {
        return district;
    }

    public void setDistrict(String district) {
        this.district = district;
    }

    public String getIntro() {
        return intro;
    }

    public void setIntro(String intro) {
        this.intro = intro;
    }

    public String getChineseName() {
        return chineseName;
    }

    public void setChineseName(String chineseName) {
        this.chineseName = chineseName;
    }
}
